src/Tempest/Log/src/FileHandlers/RotatingFileHandler.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Log\FileHandlers;
+
+use DateTimeImmutable;
+use InvalidArgumentException;
+use Monolog\Handler\RotatingFileHandler as MonoRotatingFileHandler;
+
+final class RotatingFileHandler extends MonoRotatingFileHandler
+{
+    public const string FILE_PER_WEEK = 'Y-W';
+
+    protected function setDateFormat(string $dateFormat): void
+    {
+        if (0 === preg_match('{^[Yy](([/_.-]?m([/_.-]?d)?)|([/_.-]?W))?$}', $dateFormat)) {
+            throw new InvalidArgumentException(
+                'Invalid date format - format must be one of '.
+                'RotatingFileHandler::FILE_PER_DAY ("Y-m-d"), RotatingFileHandler::FILE_PER_WEEK ("Y-W"), '.
+                'RotatingFileHandler::FILE_PER_MONTH ("Y-m") or RotatingFileHandler::FILE_PER_YEAR ("Y"), '.
+                'or you can set one of the date formats using slashes, underscores and/or dots instead of dashes.'
+            );
+        }
+
+        $this->dateFormat = $dateFormat;
+    }
+
+    protected function getNextRotation(): DateTimeImmutable
+    {
+        return match (str_replace(['/','_','.'], '-', $this->dateFormat)) {
+            self::FILE_PER_WEEK => (new DateTimeImmutable('first day of next week'))->setTime(0, 0, 0),
+            self::FILE_PER_MONTH => (new DateTimeImmutable('first day of next month'))->setTime(0, 0, 0),
```

Comment:
Where are these other constants?

Reply:
They are defined at the top of the base `Monolog\Handler\RotatingFileHandler`

I wanted to create a minimal invasive "wrapper" over `Monolog\Handler\RotatingFileHandler` that just adds support for weekly

<img width="704" alt="image" src="https://github.com/user-attachments/assets/802b0a52-a6ae-45f5-9321-d3221a8e8059">

Reply:
Yeah, totally missed that this class was extending monolog 😁